components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationSharingService.java

```diff
@@ -140,6 +234,181 @@ public Response shareOrganizationApplication(String applicationId,
         }
     }
 
+    /**
+     * Share an application with selected organizations.
+     *
+     * @param applicationShareSelectedRequestBody Request body of the share request.
+     * @return The status of the operation.
+     */
+    public Response shareOrganizationApplication(ApplicationShareSelectedRequestBody
+                                                         applicationShareSelectedRequestBody) {
+
+        String applicationId = applicationShareSelectedRequestBody.getApplicationId();
+        String ownerOrganizationId = getOrganizationId();
+        List<OrgShareConfig> organizations = applicationShareSelectedRequestBody.getOrganizations();
+        if (CollectionUtils.isEmpty(organizations)) {
+            throw Utils.buildClientError(ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY.getCode(),
+                    ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY.getMessage(),
+                    "No organizations details found.");
```

Comment:
`can't we use the ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY.getDescription()`

Reply:
Does it due to previous description does not align with the new endpoints?